### PR TITLE
Setup yarn audit

### LINKED ISSUE #411: Setup `yarn audit`
> - Also a built in [yarn audit](https://github.com/MetaMask/metamask-extension/blob/develop/.circleci/scripts/yarn-audit.sh) seems like a good idea.

_Originally posted by @pavlovcik in https://github.com/ubiquity/ubiquity-dollar/discussions/197#discussioncomment-3641178_

### PR DESCRIPTION
**Pull request by FibrinLab**
Resolves #411

### CODE REVIEW

File: .github/workflows/yarn-audit-actions.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+
+      - name: Install Yarn
+        run: yarn install
+
+      - name: Install Yarn Audit Package
+        run: yarn add yarn-audit-fix -D
```
Comment (0x4007) [inline .github/workflows/yarn-audit-actions.yml]:
```suggestion
```

File: .github/workflows/yarn-audit-actions.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Audit and Attempt Fix
+        run: yarn run yarn-audit-fix
```
Comment (0x4007) [inline .github/workflows/yarn-audit-actions.yml]:
```suggestion
        run: yes | npx yarn-audit-fix
```

Comment (0x4007) [reply .github/workflows/yarn-audit-actions.yml]: Saves a couple steps

File: package.json [hunk 3]
```diff
@@ -19,9 +19,11 @@
     "eslint-plugin-prettier": "^4.2.1",
     "eslint-plugin-tsc": "^2.0.0",
     "husky": "^8.0.2",
+    "improved-yarn-audit": "^3.0.0",
```
Comment (0x4007) [inline package.json]: Not sure if necessary else remove all the other changes (lockfile, and added deps)

**Review by 0x4007 — CHANGES_REQUESTED** [review]

File: .github/workflows/yarn-audit-actions.yml (hunk 1 above)
Comment (FibrinLab) [reply .github/workflows/yarn-audit-actions.yml]: Modified

File: .github/workflows/yarn-audit-actions.yml (hunk 2 above)
Comment (FibrinLab) [reply .github/workflows/yarn-audit-actions.yml]: Modified. Working now

File: package.json (hunk 3 above)
Comment (FibrinLab) [reply package.json]: Redundant dependency removed

**Comment by FibrinLab** [comment]
Requesting another review

**Review by FibrinLab — COMMENTED** [review]
Changes implemented

**Comment by 0x4007** [comment]
lol I just manually solved the merge conflict for yarn.lock this seems messy.

**Pull request merged**